lib/src/widgets/overlay.dart

```diff
@@ -230,15 +202,42 @@ class _DescribedFeatureOverlayState extends State<DescribedFeatureOverlay>
     super.didChangeDependencies();
   }
 
-  void _setDismissStream(Stream<void> newStream) {
-    _dismissStreamSubscription?.cancel();
-    _dismissStream = newStream;
-    _dismissStreamSubscription =
-        _dismissStream.listen((String featureId) async {
+  @override
+  void dispose() {
+    _openController.dispose();
+    _completeController.dispose();
+    _dismissController.dispose();
+    _pulseController.dispose();
+
+    // If this widget is disposed while still showing an overlay,
+    // it needs to remove itself from the active overlays.
+    //
+    // This is not done when closing the overlay because the Bloc
+    // resets the activeOverlays and this would interfere with that.
+    //
+    // Dismissing and activating are not considered "showing" in this case
+    // because the Bloc will already have dealt with the activeOverlays as
+    // it triggered the completion or dismissal animation.
+    if (_state != FeatureOverlayState.closed &&
+        _state != FeatureOverlayState.dismissing &&
+        _state != FeatureOverlayState.activating) {
+      // If the _state is anything else, this overlay has to be showing,
+      // otherwise something is wrong.
+      assert(bloc.activeFeatureId == widget.featureId);
+
+      bloc.activeOverlays--;
+    }
+    super.dispose();
+  }
+
+  void _setStartStream(Stream<void> newStream) {
```

Comment:
This might seem like I changed a lot here, but I actually didn't. In fact , I did not really change any functionality to be precise.

I will walk you through what I did: I wanted to increase consistency and enforced three concepts:

 * All of the overridden `State` methods should be at the top of the class. Most of them already were, but `dispose` was not.

 * The order of the three major actions (open, complete, dismiss) should always be the same for orientation purposes, i.e. first open (I renamed your start streams in a later commit), then complete, and then commit. This now helps you (or at least me) keep orientation in that class because the streams, animation controllers, and methods are all organized in that order.

 * All of the `_state` assignments should be in the action methods: `_open` (I moved `_show` into `_open` later, for consistency, again) etc. and not in the animation status listeners.

You can [view the full commit here](https://github.com/axel-op/feature_discovery/pull/38/commits/4bf311829ee6e501c611da1579c164104a2c1d75).